src/sparql/taxon-range-violation.sparql

```diff
@@ -0,0 +1,12 @@
+PREFIX owl: <http://www.w3.org/2002/07/owl#>
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+PREFIX never_in_taxon: <http://purl.obolibrary.org/obo/RO_0002161>
+PREFIX present_in_taxon: <http://purl.obolibrary.org/obo/RO_0002175>
+
+SELECT ?term ?taxon
+WHERE {
+  ?term never_in_taxon:|present_in_taxon: ?taxon .
+
+  FILTER (!isIRI(?taxon) && (!STRSTARTS(STR(?taxon), "http://purl.obolibrary.org/obo/NCBITaxon_")))
```

Comment:
should be:
!isIRI(?taxon) || (isIRI(?taxon) && ((!STRSTARTS(STR(?taxon), "http://purl.obolibrary.org/obo/NCBITaxon_")))

Reply:
@shawntanzk with Ubergraph you can specify particular named graphs to limit which inferences are returned. Also, much faster to use `?term rdfs:isDefinedBy <http://purl.obolibrary.org/obo/bfo.owl>` instead of FILTER. But these tips are only if you are querying only Ubergraph.

Reply:
See https://github.com/INCATools/ubergraph#graph-organization

Reply:
@shawntanzk thanks, but no need to verify again if `isIRI` on the second side :)
`!isIRI(?taxon) || !STRSTARTS(STR(?taxon), "http://purl.obolibrary.org/obo/NCBITaxon_")`

Reply:
@anitacaron yeah I guess makes sense, if it starts with "http://purl.obolibrary.org/obo/" it is a IRI lol

Reply:
Are you sure? What if it's a string? I think you need isIRI. What makes sure that it is an iri rather than a string looking like an IRI?

Reply:
It will fail in the first part `!isIRI(?taxon) || !STRSTARTS(STR(?taxon), "http://purl.obolibrary.org/obo/NCBITaxon_")`